web/src/components/storage/LvmPage.test.tsx

```diff
@@ -0,0 +1,300 @@
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { screen, within } from "@testing-library/react";
+import { installerRender, mockParams } from "~/test-utils";
+import { model, StorageDevice } from "~/types/storage";
+import { apiModel } from "~/api/storage/types";
+import { gib } from "./utils";
+import { Drive } from "~/types/storage/model";
+import LvmPage from "./LvmPage";
+
+const sda1: StorageDevice = {
+  sid: 69,
+  name: "/dev/sda1",
+  description: "Swap partition",
+  isDrive: false,
+  type: "partition",
+  size: gib(2),
+  shrinking: { unsupported: ["Resizing is not supported"] },
+  start: 1,
+};
+
+const sda: StorageDevice = {
+  sid: 59,
+  isDrive: true,
+  type: "disk",
+  vendor: "Micron",
+  model: "Micron 1100 SATA",
+  driver: ["ahci", "mmcblk"],
+  bus: "IDE",
+  busId: "",
+  transport: "usb",
+  dellBOSS: false,
+  sdCard: true,
+  active: true,
+  name: "/dev/sda",
+  size: 1024,
+  shrinking: { unsupported: ["Resizing is not supported"] },
+  systems: [],
+  partitionTable: {
+    type: "gpt",
+    partitions: [sda1],
+    unpartitionedSize: 0,
+    unusedSlots: [{ start: 3, size: gib(2) }],
+  },
+  udevIds: ["ata-Micron_1100_SATA_512GB_12563", "scsi-0ATA_Micron_1100_SATA_512GB"],
+  udevPaths: ["pci-0000:00-12", "pci-0000:00-12-ata"],
+  description: "",
+};
+
+const mockSdaDrive: Drive = {
+  name: "/dev/sda",
+  spacePolicy: "delete",
+  partitions: [
+    {
+      mountPath: "swap",
+      size: {
+        min: gib(2),
+        default: false, // false: user provided, true: calculated
+      },
+      filesystem: { default: false, type: "swap" },
+    },
+    {
+      mountPath: "/home",
+      size: {
+        min: gib(16),
+        default: true,
+      },
+      filesystem: { default: false, type: "xfs" },
+    },
+  ],
+  isUsed: true,
+  getVolumeGroups: () => [],
+};
+
+const mockRootVolumeGroup: model.VolumeGroup = {
+  vgName: "fakeRootVg",
+  getTargetDevices: () => [mockSdaDrive],
+  logicalVolumes: [],
+};
+
+const mockAddVolumeGroup = jest.fn();
+const mockEditVolumeGroup = jest.fn();
+let mockVolumeGroups: apiModel.VolumeGroup[] = [];
+
+let mockUseModel = {
+  drives: [mockSdaDrive],
+  volumeGroups: mockVolumeGroups,
+};
+
+const mockUseAllDevices = [sda];
+
+jest.mock("~/queries/issues", () => ({
+  ...jest.requireActual("~/queries/issues"),
+  useIssuesChanges: jest.fn(),
+  useIssues: () => [],
+}));
+
+jest.mock("~/queries/storage", () => ({
+  ...jest.requireActual("~/queries/storage"),
+  useAvailableDevices: () => mockUseAllDevices,
+  useDevices: () => [sda],
+}));
+
+jest.mock("~/hooks/storage/model", () => ({
+  ...jest.requireActual("~/hooks/storage/model"),
+  __esModule: true,
+  useVolumeGroup: (id: string) => (id ? mockRootVolumeGroup : null),
+  default: () => mockUseModel,
```

Comment:
A _side effect_ of [_exporting these hooks as default_](https://github.com/agama-project/agama/pull/2123#discussion_r2006544612) :man_shrugging: